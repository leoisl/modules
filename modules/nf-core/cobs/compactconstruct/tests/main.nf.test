nextflow_process {
    name "Test Process COBS_COMPACTCONSTRUCT"
    script "../main.nf"
    process "COBS_COMPACTCONSTRUCT"

    tag "modules"
    tag "modules_nfcore"
    tag "cobs"
    tag "cobs/compactconstruct"

    test("sarscov2 - genome - fasta") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert process.out.index != null },
                { assert process.out.index.get(0).get(1) ==~ ".*/index.cobs_compact" },
                { assert snapshot(process.out.index).match("index")},
                { assert snapshot(process.out.versions).match("versions") }
            )
        }
    }
}
