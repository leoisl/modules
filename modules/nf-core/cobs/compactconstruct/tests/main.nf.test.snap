{
    "versions": {
        "content": [
            [
                "versions.yml:md5,d8ab36dd4f54e5b1c34405e330774cfb"
            ]
        ],
        "timestamp": "2023-12-06T19:25:51.729793"
    },
    "index": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "index.cobs_compact:md5,56d89956e4c2cb0f5994af9720f2d882"
                ]
            ]
        ],
        "timestamp": "2023-12-06T19:25:51.705329"
    }
}